### PR TITLE
Modinvstor 391 effective call number migration script

### PR DESCRIPTION
**Pull request by rladdusaw**
- Created a migration script.
- Created tests to verify correctness.

### CODE REVIEW

File: src/main/resources/templates/db_scripts/populateEffectiveCallNumberComponentsForExistingItems.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+UPDATE ${myuniversity}_${mymodule}.item AS it
```
Comment (bohdan-suprun) [inline src/main/resources/templates/db_scripts/populateEffectiveCallNumberComponentsForExistingItems.sql]: We have to include this file into schema.json file (see `script` section)

File: src/main/resources/templates/db_scripts/populateEffectiveCallNumberComponentsForExistingItems.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+SET jsonb =
+  CASE
+    WHEN it.jsonb->'itemLevelCallNumber' IS NOT NULL
+      THEN
+        CASE
+          WHEN it.jsonb->'effectiveCallNumberComponents' IS NULL THEN
+            jsonb_set(it.jsonb, '{effectiveCallNumberComponents}',
+              jsonb_set('{"callNumber":null}', '{callNumber}', it.jsonb->'itemLevelCallNumber'))
```
Comment (bohdan-suprun) [inline src/main/resources/templates/db_scripts/populateEffectiveCallNumberComponentsForExistingItems.sql]: There is also `json_build_object` function, that can be used to build an object:


Example | Example Result
-- | --
json_build_object('foo',1,'bar',2) | {"foo": 1, "bar": 2}

**Comment by rladdusaw** [comment]
@bohdan-suprun I've updated my PR with the changes you requested.

**Comment by marcjohnson-kint** [comment]
> Thanks, looks good to me.
> 
> I think we will have to restructure a bit the migration file in future, since introducing of new properties will be reducing file maintainability.

When sortable call number is introduced, it is likely we will need Java based migration as well.

**Pull request merged**